controllers/storagecluster/reconcile.go

```diff
@@ -402,9 +404,13 @@ func (r *StorageClusterReconciler) reconcilePhases(
 			&ocsExternalResources{},
 			&ocsStorageQuota{},
 			&ocsCephCluster{},
-			&ocsSnapshotClass{},
 			&ocsNoobaaSystem{},
 		}
+
+		// do not create default snapshotClass in consumer mode
+		if !IsOCSConsumerMode(instance) {
+			objs = append(objs, &ocsSnapshotClass{})
+		}
```

Comment:
I think this is a bad idea, trying to dynamically create the content of the invoked list.
A better and a much simpler solution is to always add `ocsSnapshotClass` to the list and to have a guard at the start of the `ensure*`  function of the form:
```
if !IsOCSConsumerMode(instance) { 
    return nil
}
```

Reply:
done